tests/cli/test_cli.py

```diff
@@ -0,0 +1,505 @@
+    create_example_project(tmp_path)
+
+    # Example project models have start dates, so there are no date prompts
+    # for the `prod` environment. User input `y` is required to apply the plan.
+    result = runner.invoke(cli, ["--paths", tmp_path, "plan"], input="y\n")
+    assert_plan_success(result)
+
+
+def test_plan_skip_tests(runner, tmp_path):
+    create_example_project(tmp_path)
+
+    # Successful test run message should not appear if `--skip-tests` is passed and plan is applied
+    result = runner.invoke(cli, ["--paths", tmp_path, "plan", "--skip-tests"], input="y\n")
+    assert result.exit_code == 0
+    assert "Successfully Ran 1 tests against duckdb" not in result.output
+    assert_new_env(result)
+    assert_backfill_success(result)
+
+
+def test_plan_restate_model(runner, tmp_path):
+    create_example_project(tmp_path)
+
+    # Create prod and backfill
+    result = runner.invoke(cli, ["--paths", tmp_path, "plan", "--auto-apply"])
+
+    # plan with no changes and full_model restated
+    result = runner.invoke(
+        cli,
+        ["--paths", tmp_path, "plan", "--restate-model", "sqlmesh_example.full_model"],
+        input="\n\ny\n",
+    )
+    assert result.exit_code == 0
+    assert_duckdb_test(result)
+    assert "No differences when compared to `prod`" in result.output
+    assert "sqlmesh_example.full_model evaluated in" in result.output
+    assert_backfill_success(result)
+
+
+def test_plan_skip_backfill(runner, tmp_path):
+    create_example_project(tmp_path)
+
+    # plan for `prod` errors if `--skip-backfill` is passed without --no-gaps
+    result = runner.invoke(cli, ["--paths", tmp_path, "plan", "--skip-backfill"])
+    assert result.exit_code == 1
+    assert (
+        "Error: When targeting the production environment either the backfill should not be skipped or the lack of data gaps should be enforced (--no-gaps flag)."
+        in result.output
+    )
+
+    # plan executes virtual update without executing model batches
+    result = runner.invoke(
+        cli, ["--paths", tmp_path, "plan", "--skip-backfill", "--no-gaps"], input="y\n"
+    )
+    assert result.exit_code == 0
+    assert_virtual_update(result)
+    assert "All model batches have been executed successfully" not in result.output
+
+
+def test_plan_auto_apply(runner, tmp_path):
+    create_example_project(tmp_path)
+
+    # plan for `prod` runs end-to-end with no user input if `--auto-apply` is passed
+    result = runner.invoke(cli, ["--paths", tmp_path, "plan", "--auto-apply"])
+    assert_plan_success(result)
+
+
+def test_plan_verbose(runner, tmp_path):
+    create_example_project(tmp_path)
+
+    result = runner.invoke(cli, ["--paths", tmp_path, "plan", "--verbose"], input="y\n")
+    assert_plan_success(result)
+    assert "sqlmesh_example.seed_model created" in result.output
+    assert "sqlmesh_example.seed_model evaluated in" in result.output
+    assert "sqlmesh_example.seed_model promoted" in result.output
```

Comment:
Want to assert that these do not show up in the test above this? Right now these could always show up and not fail a test.